src/rapids_singlecell/pertpy_gpu/_metrics/_edistance.py

```
@@ -327,14 +347,15 @@ def onesided_distances(
             # e[s,b] = 2*d[s,b] - d[s,s] - d[b,b]
             ed_cols = {}
             var_cols = {}
-            for i, (sg, si) in enumerate(zip(selected_groups, selected_indices)):
-                ed_row = 2 * cross_mean[i, :] - diag_mean[si] - diag_mean
-                ed_row[si] = 0.0
-                ed_cols[sg] = ed_row.get()
+            with cp.cuda.Device(source_device):
+                for i, (sg, si) in enumerate(zip(selected_groups, selected_indices)):
```

Ruff (0.16.2)

[warning] `zip()` without an explicit `strict=` parameter

Add explicit value for parameter `strict=`

(B905)